### PR TITLE
REFACTOR-#4047: Rename `cluster` directory to `cloud` in examples

### LINKED ISSUE #4047: Rename `cluster` directory to `cloud` in examples
We should rename `cluster` directory to `cloud` in examples because the original name of the showcased feature is Modin in the cloud. Also, we will likely want to create separate directories `local` and `cluster` in the future where we show Ray and Dask to be run.

cc @vnlitvinov, @anmyachev

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4047  
- [x] tests passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: docs/release_notes/release_notes-0.14.0.rst [hunk 1]
```diff
@@ -18,6 +18,7 @@ Key Features and Updates
 * Refactor Codebase
   * REFACTOR-#3990: remove code duplication in `PandasDataframePartition` hierarchy (#3991)
   * REFACTOR-#3997: remove code duplication for `broadcast_apply` method (#3996)
+  * REFACTOR-#4047: Rename `cluster` directory to `cloud` in examples
```
Comment (YarShev) [inline docs/release_notes/release_notes-0.14.0.rst:21]:
```suggestion
  * REFACTOR-#4047: Rename `cluster` directory to `cloud` in examples (#4212)
```

**Comment by YarShev** [comment]
> @YarShev please clear conflicts, otherwise LGTM

done

**Comment by YarShev** [comment]
@devin-petersohn, ping

**Review by devin-petersohn — COMMENTED** [review]
Should this be `orchestration` instead of `cloud`? I think that name might make more sense.

**Comment by YarShev** [comment]
I think it would probably be a better name for the feature than Modin in the Cloud. We should rename the feature as a whole as part of a separate issue if we really want. For now I think renaming `cluster` to `cloud` should land as is.

**Review by devin-petersohn — APPROVED** [review]
Thanks @YarShev , let's do the rename in another PR

**Pull request merged**